### PR TITLE
Update SMMP GPT sample

### PR DESCRIPTION
**Pull request by haohanchen-yagao**
*Issue #, if available:*

*Description of changes: Update SMMP GPT2 script with the latest changes, the main changes are:

- Removed fp16 folder as fp16 related functions/wrappers are integrated into SMMP code base now. 
- Updated the memory tracker
- Removed private functions/arguments from public example
- Update SMMP APIs in training scripts
- Updated transformer version
- Updated Pytorch version to 1.12 in notebooks

Note: This PR shouldn't be merged until changes to SMMP is released.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [ ] I have linted my notebook(s) and code using `tox -e black-format,black-nb-format`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by YangFei1990 — DISMISSED** [review]

File: training/distributed_training/pytorch/model_parallel/gpt-j/11_train_gptj_smp_tensor_parallel_notebook.ipynb [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "! pip install -qU \"sagemaker>=2,<3\"\n",
     "! pip install -qU sagemaker-experiments\n",
     "\n",
-    "# import IPython\n",
+    "# import IPythonfp16\n",
```
Comment (mchoi8739) [inline training/distributed_training/pytorch/model_parallel/gpt-j/11_train_gptj_smp_tensor_parallel_notebook.ipynb:42]:
```suggestion
    "# import IPython\n",
```

File: training/distributed_training/pytorch/model_parallel/gpt2/smp-train-gpt-simple.ipynb [hunk 2]
```diff
@@ -13,16 +13,17 @@
...
     "This notebook depends on the following files and folders:\n",
     "\n",
     "- `train_gpt_simple.py`: This is an entrypoint script that is passed to the Hugging Face estimator in the notebook instructions. This script is responsible for end to end training of the GPT-2 model with SMP. The script has additional comments at places where the SMP API is used.\n",
-    "- `fp16`: This folder is used for 16-bit float training, which contains a fp16 optimizer and various fp16 utilities.\n",
     "- `data_pipeline.py`: This contains the datapipeline function to prepare the training data.\n",
     "- `learining_rate.py`: This contains the functions for learning rate schedule.\n",
     "- `requirements.txt`: This will install the dependencies, like the right version of huggingface transformers.\n",
     "- `data_prep_512.py`: This will download and preprocess the openwebtext dataset.\n",
+    "- `memory_tracker.py`: This contains the functions to track memory usage.\n",
+    "- `sharded_data_parallel_checkpoint.py`: This contains the checkpoint utils for sharded dat parallelism\n",
```
Comment (mchoi8739) [inline training/distributed_training/pytorch/model_parallel/gpt2/smp-train-gpt-simple.ipynb:26]:
```suggestion
    "- `sharded_data_parallel_checkpoint.py`: This contains checkpoint util functions for sharded data parallelism.\n",
```

**Review by mchoi8739 — CHANGES_REQUESTED** [review]

File: training/distributed_training/pytorch/model_parallel/gpt-j/11_train_gptj_smp_tensor_parallel_notebook.ipynb [hunk 3]
```diff
@@ -39,7 +39,7 @@
     "! pip install -qU \"sagemaker>=2,<3\"\n",
```
Comment (mchoi8739) [inline training/distributed_training/pytorch/model_parallel/gpt-j/11_train_gptj_smp_tensor_parallel_notebook.ipynb:39]: <p>Suggest to change the intro sentence:</p><p> </p><p>You'll learn how to train the GPT-J model with tensor parallelism on the GLUE sst2 dataset.</p><p> </p><p>--&gt;</p><p> </p><p>You'll learn how to run FP16 training of the GPT-J model with tensor parallelism on the GLUE sst2 dataset.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3433/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (mchoi8739) [inline training/distributed_training/pytorch/model_parallel/gpt-j/11_train_gptj_smp_tensor_parallel_notebook.ipynb:39]: <p>Change the title to:</p><p> </p><h3>Create a SageMaker PyTorch Estimator</h3><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3433/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='54'/>

File: training/distributed_training/pytorch/model_parallel/gpt2/smp-train-gpt-simple.ipynb [hunk 4]
```diff
@@ -4,7 +4,7 @@
    "cell_type": "markdown",
```
Comment (mchoi8739) [inline training/distributed_training/pytorch/model_parallel/gpt2/smp-train-gpt-simple.ipynb:4]: <p>Same suggestion for the intro paragraph:</p><p> </p><p>You'll learn how to train the GPT-2 model with SageMaker's model parallelism.</p><p>--&gt;</p><p>You'll learn how to run FP16 training of the GPT-2 model with SageMaker's model parallelism.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3433/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (mchoi8739) [inline training/distributed_training/pytorch/model_parallel/gpt2/smp-train-gpt-simple.ipynb:4]: <p>Change HuggingFace to PyTorch in the title and the text.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3433/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

**Comment by mchoi8739** [comment]
Run `black-nb -l 100` to pass the code formatting test.

**Review by mchoi8739 — APPROVED** [review]

**Review by YangFei1990 — APPROVED** [review]

**Review by aaronmarkham — APPROVED** [review]

**Pull request merged**